packages/@react-spectrum/list/src/ListViewItem.tsx

```diff
@@ -94,19 +112,41 @@ export function ListViewItem(props) {
               'is-hovered': isHovered,
               'is-selected': isSelected,
               'is-previous-selected': state.selectionManager.isSelected(item.prevKey),
-              'react-spectrum-ListViewItem--highlightSelection': state.selectionManager.selectionBehavior === 'replace' && (isSelected || state.selectionManager.isSelected(item.nextKey))
+              'react-spectrum-ListViewItem--highlightSelection': state.selectionManager.selectionBehavior === 'replace' && (isSelected || state.selectionManager.isSelected(item.nextKey)),
+              'react-spectrum-ListViewItem--draggable': isDraggable
             }
           )
         }
         ref={cellRef}
         {...mergedProps}>
         <Grid UNSAFE_className={listStyles['react-spectrum-ListViewItem-grid']}>
-          {showCheckbox && (
+          {isListDraggable &&
+            <div className={listStyles['react-spectrum-ListViewItem-draghandle-container']}>
+              {isDraggable &&
+                <FocusRing focusRingClass={classNames(listStyles, 'focus-ring')}>
+                  <div
+                    {...buttonProps as React.HTMLAttributes<HTMLElement>}
+                    className={
+                      classNames(
+                        listStyles,
+                        'react-spectrum-ListViewItem-draghandle-button'
+                      )
+                    }
+                    style={!showDragHandle ? {...visuallyHiddenProps.style} : {}}
```

Comment:
Using the visuallyHiddenStyles to make the drag handle hidden when the row isn't focused/pressed/hovered so that screenreaders still have access to the button. Open to opinions, an alternative would be to have the handle be visible at all times.

Reply:
I think that is the best solution.

Reply:
visually hidden seems best